packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -67,6 +70,60 @@ import { SchemaParser } from "./schemaParser";
 import { InMemoryNoteCache } from "../../util/inMemoryNoteCache";
 import { NotesFileSystemCache } from "../../cache";
 
+export enum EngineInitErrorType {
+  DUPLICATE_NOTE_ID = "duplicate note id",
+}
+
+export class DuplicateNoteError extends BaseDendronError<EngineInitErrorType.DUPLICATE_NOTE_ID> {
+  translateToHTTPErrorCode(): StatusCodes | undefined {
+    return StatusCodes.INTERNAL_SERVER_ERROR;
+  }
+
+  constructor(
+    opts: Omit<
+      DendronErrorProps<EngineInitErrorType.DUPLICATE_NOTE_ID>,
+      "name" | "message" | "severity"
+    > & {
+      noteA: NoteProps;
+      noteB: NoteProps;
+    }
+  ) {
+    super({
+      ...opts,
+      severity: ERROR_SEVERITY.MINOR,
+      message: `Notes ${opts.noteA.fname} in ${VaultUtils.getName(
+        opts.noteA.vault
+      )} and ${opts.noteB.fname} in ${VaultUtils.getName(
+        opts.noteB.vault
+      )} have duplicate IDs.`,
+      code: EngineInitErrorType.DUPLICATE_NOTE_ID,
+    });
+    this.noteA = {
+      fname: opts.noteA.fname,
+      vault: opts.noteA.vault,
+    };
+    this.noteB = {
+      fname: opts.noteB.fname,
+      vault: opts.noteB.vault,
+    };
+  }
+
+  public noteA: {
```

Comment:
rather than noteA/noteB, does it make more sense to have a property of type {fname, vault} array? In case there are 3 notes with the same ID for example.

Reply:
The way it works right now can't detect more than a pair of duplicate note IDs. If there are 3 notes with the same ID then there would be 2 different errors. I'll leave it as-is for now, if someone implements a better check that can capture multi-note conflicts then they can refactor it.